Replace YYPARSE_PARAM (which Bison 3 no longer supports) with
%parse-param and update function prototypes as necessary.
Index: nasl/nasl_grammar.y
===================================================================
--- nasl/nasl_grammar.y.orig
+++ nasl/nasl_grammar.y
@@ -1,4 +1,5 @@
-%pure_parser
+%define api.pure
+%param {void * parm}
 %expect 1
 %{
 /* Nessus Attack Scripting Language version 2
@@ -19,9 +20,6 @@
  * Foundation, Inc., 675 Mass Ave, Cambridge, MA 02139, USA.
  */
 
-#define YYPARSE_PARAM parm
-#define YYLEX_PARAM parm
-
 #define LNB	(((naslctxt*)parm)->line_nb)
 #include "includes.h"
 #include "nasl_tree.h"
@@ -32,7 +30,7 @@
 #include "nasl_debug.h"
 #include "nasl_signature.h"
 
-static void naslerror(const char *);
+static void naslerror(void *, const char *);
 #define YYERROR_VERBOSE
 %}
 
@@ -470,7 +468,7 @@ glob: GLOBAL arg_decl 
 #include <stdlib.h>
 
 static void 
-naslerror(const char *s)
+naslerror(void *parm, const char *s)
 {
   fputs(s, stderr);
 }
